Standards/scs-XXXX-vN-decision-record-template.md

```diff
@@ -0,0 +1,50 @@
+---
```

Comment:
Another thing that needs to be discussed is the structure of this DR template.

I tried to derive it from the ADR templates found here https://github.com/joelparkerhenderson/architecture-decision-record/tree/main/locales/en/templates, but there are many different ideas on the web how a DR should be handled. Some examples:
https://docs.aws.amazon.com/prescriptive-guidance/latest/architectural-decision-records/adr-process.html
https://www.ozimmer.ch/practices/2022/11/22/MADRTemplatePrimer.html
http://www.iso-architecture.org/42010/templates/42010-ad-template.pdf

I wouldn't put myself out there to say what is right or wrong. The gist I collected from reading multiple articles is that the following information should be included:
* What issue is the DR deciding on and why?
* What options were considered and why?
* What is the final decision?
* How does this affect the product/current status (positive and negative)?
* (Who was involved in the decision and when was it done?)

I'm up for a discussion on this, if someone else has a stronger opinion here, but I also want to say that this whole topic is probably very much opinionated. I only I would write a DR, I would just include the actual decision and the reasoning behind it, which would also include an explanation why certain options where not picked. This is enough IMO, since certain topics can become very hard to unravel and someone could always ask "Why didn't you do X?". Time is always a factor and also with DRs (or standards), you can only look at a certain number of options before making a decision, which is why an explanation for not picking some option isn't really always that useful. A good explanation on why some option was chosen makes more sense here.

Reply:
After further thinking about this, I would maybe only add one additional optional bullet point to the above list:

* What options were intentionally not considered and why?

This should only be used when it makes sense to do so, e.g. "we didn't consider option X because the time to investigate it was $foobar, also other $reason".

Maybe this doesn't need to be spelled out, but I often find myself reading documentation and wondering why option X was never considered and when asking people it turns out it was not considered for good reasons - so someone thought about that already - but this was never spelled out anywhere. I recognize though that of course you can't document every option you didn't consider, so this only makes sense for very prominent options, that come to mind immediately.